src/Cis194/Hw/LogAnalysis.hs

```diff
@@ -6,22 +6,44 @@ module Cis194.Hw.LogAnalysis where
 import  Cis194.Hw.Log
 
 parseMessage :: String -> LogMessage
-parseMessage s = Unknown s
+parseMessage s =
+  case words s of
+    "E":i:ws -> buildMessage (Error (read i)) ws
```

Comment:
`read` can be OK when you're sure that the input string will parse, or when you're doing some temporary debugging code, but if you want to detect parse failures check out `Text.Read.readMaybe :: Read a => String -> Maybe a`